rust/src/editor.rs

```diff
@@ -552,19 +567,24 @@ impl Editor {
 
     fn debug_rewrap(&mut self) {
         self.view.rewrap(&self.text, 72);
-        self.dirty = true;
+        self.view_dirty = true;
     }
 
     fn debug_test_fg_spans(&mut self) {
         print_err!("setting fg spans");
         self.view.set_test_fg_spans();
-        self.dirty = true;
+        self.view_dirty = true;
     }
 
-    fn debug_run_plugin(&mut self, tab_ctx: &TabCtx) {
+    fn debug_run_plugin(&mut self) {
         print_err!("running plugin");
-        let plugin_ctx = tab_ctx.to_plugin_ctx();
-        start_plugin(plugin_ctx);
+        let tab_ctx = self.tab_ref.lock().unwrap();
+        start_plugin(tab_ctx.get_editor_ref());
```

Comment:
If I haven't missed anything, this is the only position where the `editor_ref` from the `TabCtx` is used. In this case, the editor could maybe manage its `Arc<Mutex<Editor>>` on its own and the `editor_ref` from `TabCtx` could be removed?